XenAdmin/SettingsPanels/CPUMemoryEditPage.cs

```diff
@@ -460,28 +462,50 @@ private void ShowMemError(bool showAlways, bool testValue)
 
         private void comboBoxVCPUs_SelectedIndexChanged(object sender, EventArgs e)
         {
-            ShowVcpuError(false, true);
-            comboBoxTopology.Update((long)comboBoxVCPUs.SelectedItem);
             ValidateVCPUSettings();
+            comboBoxTopology.Update((long)comboBoxVCPUs.SelectedItem);
+            ValidateTopologySettings();
             RefreshCurrentVCPUs();
         }
 
-        private void ShowVcpuError(bool showAlways, bool testValue)
+        private void ValidateVCPUSettings()
         {
             if (vm == null || !comboBoxVCPUs.Enabled)
                 return;
             Host selectedAffinity = vm.Home();
             if (selectedAffinity == null && vm.Connection.Cache.Hosts.Length == 1)
                 selectedAffinity = vm.Connection.Cache.Hosts[0];
 
-            if (showAlways || (testValue && (selectedAffinity != null && selectedAffinity.host_CPUs.Count < (long)comboBoxVCPUs.SelectedItem)))
+            if (selectedAffinity != null && selectedAffinity.host_CPUs.Count < (long)comboBoxVCPUs.SelectedItem)

```

Comment:
Judging by the null check further down, it is presumed likely that the SelectedItem might be null, but here we cast it directly to long, shall we add a check?